apps/cowswap-frontend/src/modules/tradeQuote/utils/quoteUsingSameParameters.ts

```diff
@@ -50,6 +72,20 @@ function compareAppDataWithoutQuoteData(a: AppDataInfo['doc'] | undefined, b: Ap
     return a === b
   }
 
+  // When environment is set in one appData and not in the other, we need to remove it
```

Comment:
What I don't get is, why/when is it not set?

Reply:
🤦 
Sorry, I found and fixed the cause:
https://github.com/cowprotocol/cow-sdk/pull/294/commits/9024a44c18f5c7919d6e948282391e975b3ac54e

Thanks for insisting on it!